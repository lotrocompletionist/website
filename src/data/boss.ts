export interface IBoss {
  id: number;
  name: string;
}
