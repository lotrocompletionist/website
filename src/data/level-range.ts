export interface ILevelRange {
  minimum: number;
  maximum?: number;
}
