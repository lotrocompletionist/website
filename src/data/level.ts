export interface Level {
  minimum: number;
  maximum?: number;
}
