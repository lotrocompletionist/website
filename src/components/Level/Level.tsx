import * as React from "react";
import { Level } from "../../data/models";

export interface LevelProps {
  level: Level;
}

export function Level(props: LevelProps) {
  const maximum = props.level.maximum ? " - " + props.level.maximum : "";

  return (
    <>
      Level: {props.level.minimum}
      {maximum}
    </>
  );
}
