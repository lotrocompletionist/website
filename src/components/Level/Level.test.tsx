import * as React from "react";
import { create } from "react-test-renderer";
import { Level } from "./Level";
import { ILevel } from "../../data/level";

test("<Level> renders correctly", () => {
  const level: ILevel = {
    minimum: 10,
    maximum: 20
  };
  const tree = create(<Level level={level} />).toJSON();
  expect(tree).toMatchSnapshot();
});
