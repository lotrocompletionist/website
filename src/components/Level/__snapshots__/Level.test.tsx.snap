// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Level> renders correctly 1`] = `
Array [
  "Level: ",
  "5",
]
`;

exports[`<LevelRange> renders correctly with  only mimimum specified 1`] = `null`;
